lib/assets/javascripts/cartodb/table/mapview.js

```diff
@@ -16,6 +16,10 @@ function GrouperLayerMapView(mapViewClass) {
       this.groupLayer = null;
       this.activeLayerModel = null;
       mapViewClass.prototype.initialize.call(this);
+
+      if (!this.options.readOnly) {
```

Comment:
It should be possible to apply geometric filters to a readonly table. Did this cause issues for you?